src/gwproto/data_classes/hardware_layout.py

```diff
@@ -197,9 +198,65 @@ def load_nodes(
                 errors.append(LoadError("ShNode", node_dict, e))
         return nodes
 
+    @classmethod
```

Comment:
I think I'd propose that this throw and error if the nodes don't exist, and then I dont think we need the resolve function for data channels.

Reply:
e.g..
```
    @classmethod
    def make_channel(cls, dc_dict: dict, nodes: dict[str, ShNode]) -> DataChannel:
        about_node = nodes.get(dc_dict.get("AboutNodeName"))
        captured_by_node = nodes.get(dc_dict.get("CapturedByNodeName"))
        if about_node is None or captured_by_node is None:
                raise ValueError(
                    f"ERROR. DataChannel related nodes exist.\n"
                    f"  For AboutNodeName <{dc_dict.get('AboutNodeName')}> "
                    f"got {about_node}\n"
                    f"  for CapturedByNodeName <{dc_dict.get('CapturedByNodeName')}>"
                    f"got {captured_by_node}"
                )
        return DataChannel(
            about_node=about_node, captured_by_node=captured_by_node, **dc_dict
        )    
```

Reply:
Sure. I am fine with use_enum_values. We can discuss extra="allow" - I think it is useful in certain circumstances when we know there might be extra information we want to throw into a type.